### ISSUE TITLE
Simplify /whisper usage

### ISSUE THREAD

**Issue by acerspyro** [red-eclipse/base#764]
Currently, to use `/whisper`, you need to know either the exact name of the player you're trying to whisper to or their client number. And you have to type that in every time you want to send a whisper, as there is no reply function. A lot of people don't even understand how CN's work, and it's not shown by default on the scoreboard.

By default, `/whisper` should have some aliases, here are some ideas:

- `/tell`; `/t`
- `/msg`; `/m`
- `/whisper`; `/w`

There should also be a command to reply to the last whisper you've received, something like /reply; /r

It would also be useful to have a dedicated "whisper chat", similar to how team chat and global chat are two distinct chat rooms. Please propose some ideas as to how to manage multiple whispers without using the mouse (or with, I forget if the mouse gets released while chatting)

**tl;dr**: The `/whisper` command sucks, it's hard to use. Let's make it simpler.

**Comment by IceflowRE** [comment]
What happens if more players whispers to you at the same time? How to handle the reply then?

**Comment by acerspyro** [comment]
> What happens if more players whispers to you at the same time? How to handle the reply then?

"Please propose some ideas as to how to manage multiple whispers without using the mouse (or with, I forget if the mouse gets released while chatting)"

**Comment by IceflowRE** [comment]
I was just leaving this as a thing we have to think about.

**Comment by IceflowRE** [comment]
Wouldnt #763 solve this problem a bit?

**Comment by acerspyro** [comment]
@IceflowRE Somewhat, but it's still command-line based, and there's no way for new players to know the command even exists at all

**Comment by IceflowRE** [comment]
And a hexagon/ honeycomb style area (like the circle rotation) where the player names are in and you can click who you want to whisper? and maybe highlight the players who wrote to you?

**Comment by acerspyro** [comment]
It would be easier if we had a standard chat box that didn't take one bjkfillion percent of the screen.

We could have tabs'n'shit.

**Comment by shirepirate** [comment]
you can save an invisible queue of who has whispered to you that session, and tab to shift between the chats

**Comment by ZeroKnight** [comment]
>By default, /whisper should have some aliases

I mentioned this in #55, but it was never followed up on. What approach should be used, here? Should they be added as aliases in `setup.cfg`? Can you add aliases in the source?

>There should also be a command to reply to the last whisper you've received, something like /reply; /r

The idea crossed my mind back then as it's a feature I enjoy in other games, but for the simple "chat box" that RE has, there would be an uncomfortable edge case where someone else whispers you *just* before you send your `/r` message, causing you to send the message to the wrong person. I've seen chat boxes work around this by locking in the target after typing `/r<Space>` and/or transforming the `/r` command into the appropriate `/w` command, but I don't think RE is capable of this at present.

> It would also be useful to have a dedicated "whisper chat", similar to how team chat and global chat are two distinct chat rooms.

Not sure what you mean by this. Unless something has changed with how messages are sent that I'm not aware of, team messages are just like regular messages but sent only to clients on the matching team. Unless you're referring to how `T` and `Y` have a unique prompt? If so, that's just `inputcommand` sugar.

> but it's still command-line based, and there's no way for new players to know the command even exists at all

Well, there's the `clients` GUI which has a whisper button, but new players wont know about that unless they saw the `F8` bind in the controls menu. I think that there should be a menu item for this on the main menu when in a game (ie. make use of `isconnected`).

> you can save an invisible queue of who has whispered to you that session, and bind TAB to shift between the chats

We'd have to make tab-completion context aware to support this, which isn't a bad idea at all.

**Comment by qreeves** [comment]
UPDATE: Development version supports tab completion of player names.

This issue is still valid as I would like to support the *whisper to last whisperer* thing.
